src/main/kotlin/com/github/leomillon/uuidgenerator/parser/CUIDUtils.kt

```diff
@@ -0,0 +1,18 @@
+package com.github.leomillon.uuidgenerator.parser
+
+const val CUID_LENGTH = 25
+
+val FIND_CUID_REGEX = """[cC][a-zA-Z0-9]{24}""".toRegex()
+
+class CUIDParser(
+    private val source: String
+) {
+
+    fun isValid(): Boolean = source.length == CUID_LENGTH && (source[0] == 'c' || source[0] == 'C') && source.all {
+        it in '0'..'9' || it in 'A'..'Z' || it in 'a'..'z'
```

Comment:
In the other parsers, I try to exit as fast as possible depending on the "ID syntax".

Here as you already checked the length (so you won't do useless further checks), you could use your regex directly.

```suggestion
    fun isValid(): Boolean = source.length == CUID_LENGTH && FIND_CUID_REGEX.matchEntire(source) != null
```